cardano-cli/src/Cardano/CLI/Types/Errors/TxValidationError.hs

```diff
@@ -70,64 +59,54 @@ validateScriptSupportedInEra era script@(ScriptInAnyLang lang _) =
                         (AnyScriptLanguage lang) (anyCardanoEra $ toCardanoEra era)
     Just script' -> pure script'
 
-newtype TxTotalCollateralValidationError
-  = TxTotalCollateralNotSupported AnyCardanoEra
-  deriving Show
-
-instance Error TxTotalCollateralValidationError where
-  prettyError (TxTotalCollateralNotSupported era) =
-    "Transaction collateral not supported in " <> pretty era
-
-validateTxTotalCollateral :: CardanoEra era
+data TxNotSupportedInEraValidationError era =
+    -- | First argument is the kind of data that is not supported.
+    -- Second argument is the era that doesn't support the data.
+    TxNotSupportedInAnyCardanoEraValidationError T.Text AnyCardanoEra
```

Comment:
This constructor can actually be removed.

Reply:
Good catch, done :+1:

Reply:
It should return `AnyShelleyBasedEra` because: https://github.com/IntersectMBO/cardano-cli/pull/783/files#diff-a38343c1825e6e334224d0a8a1ba78a458652970f00c54f5fb0d040be28f822eR606

Reply:
@Jimbo4350> not really, because this constructor is passed as a function to [conjureWitness](https://github.com/IntersectMBO/cardano-cli/blob/smelc/reduce-boilerplate-modernize-TxValidationErrors/cardano-cli/src/Cardano/CLI/Types/Errors/TxValidationError.hs#L151) and there remains [one single caller](https://github.com/IntersectMBO/cardano-cli/blob/smelc/reduce-boilerplate-modernize-TxValidationErrors/cardano-cli/src/Cardano/CLI/Types/Errors/TxValidationError.hs#L146) to `conjureWitness` that uses a `CardanoEra era` value.

Once this last caller is eliminated, we will be able to tighten `TxNotSupportedInAnyCardanoEraValidationError`'s type.

Reply:
We could use `shelleyBasedToCardanoEra`.

Reply:
@Jimbo4350> but this is awkward because `conjureWitness` expects a function to build the error, to which it applies the `CardanoEra`. If I want to keep `conjureWitness` existing type (which I think is nice), I need to do something like this:

![image](https://github.com/IntersectMBO/cardano-cli/assets/634720/310185e1-45e5-4638-8aa2-42a560a8b712)

which is weird, because the value passed by `conjureWitness` gets ignored when building the error.